plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/AbstractDeltaLakePageSink.java

```diff
@@ -0,0 +1,566 @@
+import io.trino.plugin.deltalake.DataFileInfo.DataFileType;
+import io.trino.plugin.hive.FileWriter;
+import io.trino.plugin.hive.HivePartitionKey;
+import io.trino.plugin.hive.RecordFileWriter;
+import io.trino.plugin.hive.parquet.ParquetFileWriter;
+import io.trino.plugin.hive.util.HiveUtil;
+import io.trino.plugin.hive.util.HiveWriteUtils;
+import io.trino.spi.Page;
+import io.trino.spi.PageIndexer;
+import io.trino.spi.PageIndexerFactory;
+import io.trino.spi.TrinoException;
+import io.trino.spi.block.Block;
+import io.trino.spi.connector.ConnectorPageSink;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.type.TimestampWithTimeZoneType;
+import io.trino.spi.type.Type;
+import io.trino.spi.type.TypeManager;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapred.JobConf;
+import org.apache.parquet.hadoop.metadata.CompressionCodecName;
+import org.joda.time.DateTimeZone;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.concurrent.CompletableFuture;
+
+import static com.google.common.base.Verify.verify;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.airlift.slice.Slices.wrappedBuffer;
+import static io.trino.hdfs.ConfigurationUtils.toJobConf;
+import static io.trino.plugin.deltalake.DeltaLakeErrorCode.DELTA_LAKE_BAD_WRITE;
+import static io.trino.plugin.deltalake.DeltaLakeSchemaProperties.buildHiveSchema;
+import static io.trino.plugin.deltalake.DeltaLakeSessionProperties.getCompressionCodec;
+import static io.trino.plugin.deltalake.DeltaLakeSessionProperties.getParquetWriterBlockSize;
+import static io.trino.plugin.deltalake.DeltaLakeSessionProperties.getParquetWriterPageSize;
+import static io.trino.plugin.deltalake.DeltaLakeSessionProperties.isParquetOptimizedWriterEnabled;
+import static io.trino.plugin.deltalake.transactionlog.TransactionLogAccess.canonicalizeColumnName;
+import static io.trino.plugin.hive.HiveStorageFormat.PARQUET;
+import static io.trino.plugin.hive.metastore.StorageFormat.fromHiveStorageFormat;
+import static io.trino.plugin.hive.util.CompressionConfigUtil.configureCompression;
+import static io.trino.plugin.hive.util.HiveUtil.escapePathName;
+import static io.trino.spi.type.TimestampType.TIMESTAMP_MILLIS;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static java.util.UUID.randomUUID;
+import static java.util.stream.Collectors.toList;
+
+public abstract class AbstractDeltaLakePageSink
+        implements ConnectorPageSink
+{
+    private static final Logger LOG = Logger.get(AbstractDeltaLakePageSink.class);
+
+    private static final int MAX_PAGE_POSITIONS = 4096;
```

Comment:
sure, but it was here before, I've pretty much just renamed DeltaLakePageSink. I can change it as a separate commit if you insist

Reply:
if that would be reflected in the change log, it would save me review time.

a person commit later will not know _how_ you made the change. they will only see the lines added here, and the commit title, so you will be considered the author